### PR TITLE
Final modification

### LINKED ISSUE #51: change detailed character view to be a modal

### LINKED ISSUE #54: change reset button design

### LINKED ISSUE #55: set custom color theme
Previous design:
Navigation and other components use different blacks
Screenshot 2023-04-01 at 18 42 35

After fix:
Use the same black:
Screenshot 2023-04-01 at 18 45 38

### LINKED ISSUE #56: fix the file structure
2. I found that the file structure is a bit odd. Currently, pretty much all the UI files are under `components/` but there're multiple files that are mentally bigger than "components" such as `AllSpecies.tsx`. For such a file, you might want to have `page` folder and you can put `AllSpecies.tsx` and `SpeciesItem.tsx`. Also, `UI` folder can be de-structured and put on the same level as other components. Something like the following image is what I mean, but we can have a quick call to go over this structure once you are back from Japan 🙂

Screenshot 2023-03-10 at 12 13 31 AM

_Originally posted by @kei95 in https://github.com/Lada496/star-wars-list/issues/53#issuecomment-1463433147_

### PR DESCRIPTION
**Pull request by Lada496**
#### What Feature I added
- fixed #54 
- fixed #56 
- fixed #51 
- fixed #55 : I overrode custom black

#### Others
Sorry for adding extra change in addition to re-design the reset button. I hope you don't get confused so much.
This would be the final PR for this project. Thanks a lot for your help 🙌

### CODE REVIEW

File: src/pages/AllSpecies/AllSpecies.tsx [hunk 1]
```diff
@@ -63,12 +64,39 @@ const AllSpecies = ({ characters }: AllSpeciesProps) => {
...
+      setShowClearButton(true)
+    } else {
+      setShowClearButton(false)
+    }
+  }, [
+    charactersState.sortFactor,
+    charactersState.homeworldFilterTarget,
+    charactersState.genderFilterTarget,
+  ])
+
```
Comment (kei95) [inline src/pages/AllSpecies/AllSpecies.tsx:82]: This is quite an aggressive checking you are doing 😄 I'd make a variable out of this if condition to make is a bit more readable;

```suggestion
  const shouldClearButtonPresent = charactersState.sortFactor ||
      charactersState.homeworldFilterTarget ||
      charactersState.genderFilterTarget
      
  useEffect(() => {
      setShowClearButton(shouldClearButtonPresent)
  }, [shouldClearButtonPresent])
```

File: src/components/CharacterItem/CharacterItem.tsx [hunk 2]
```diff
@@ -77,8 +76,8 @@ const CharacterItem = ({ character }: CharacterItemProps) => {
           p={6}
           textAlign={'center'}
           border={'1px'}
-          borderColor='black'
-          boxShadow={useColorModeValue('6px 6px 0 black', '6px 6px 0 cyan')}
+          borderColor='black.900'
+          boxShadow={useColorModeValue('6px 6px 0 #221F20', '6px 6px 0 cyan')}
```
Comment (kei95) [inline src/components/CharacterItem/CharacterItem.tsx:80]: I saw you added this color to a theme. Is there any particular reason why you didn't use it here?

**Review by kei95 — APPROVED** [review]
Oh man, this repository aged well 😄 It's a bit sad to see it go, but you've done so much work on this and I hope all the processes you made during the development helped you to be a better developer Yuko! I'm quite satisfied by the result we got at the end, you are not really a Jr developer when it comes to coding anymore. Good job! 👍

File: src/components/CharacterItem/CharacterItem.tsx (hunk 2 above)
Comment (Lada496) [reply src/components/CharacterItem/CharacterItem.tsx:80]: This is just because 'black.900' doesn't work with this syntax 😅

Comment (kei95) [reply src/components/CharacterItem/CharacterItem.tsx:80]: Ah kk

**Comment by Lada496** [comment]
I'll merge this branch to set this project inactive.
Again, thank you @kei95 and @koh1project for your dedicative help.

**Pull request merged**